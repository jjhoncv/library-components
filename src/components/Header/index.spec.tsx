import * as React from 'react';
import { shallow } from 'enzyme';
import { Header, HeaderImage } from './';

describe('Header Component', () => {
  describe('Header => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <Header title="Call Center" />
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Header Image => render snapshot', () => {
    it('Render Image correctly', () => {
      const wrapper = shallow(
        <HeaderImage title="Call Center" image={['https://via.placeholder.com/50', 'end']}/>
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Header Builder', () => {
    it('Header => Render prop title', () => {
      const wrapper = shallow(
        <Header title="Call Center" />
      );
      expect(wrapper.childAt(0).prop('children')).toBe('Call Center');
    });
    it('Header => Render prop subTitle', () => {
      const wrapper = shallow(
        <Header title="Call Center" subTitle="Generación de CIP" />
      );
      expect(wrapper.childAt(1).prop('children')).toBe('Generación de CIP');
    });
  });

  describe('Header Image Builder', () => {
    it('Header Image => Render prop image', () => {
      const wrapper = shallow(
        <HeaderImage title="Call Center" image={['https://via.placeholder.com/50', 'end']} />
      );
      expect(wrapper.childAt(0).prop('pos')).toBe('end');
    });
  })
});
