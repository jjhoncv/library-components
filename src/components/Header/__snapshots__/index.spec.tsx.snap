// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header => render snapshot Render correctly 1`] = `"<div class=\\"sc-bdVaJa UoSJW\\"><span class=\\"sc-bwzfXH ljAYCu\\">Call Center</span></div>"`;

exports[`Header Component Header Image => render snapshot Render Image correctly 1`] = `"<div class=\\"sc-bdVaJa UoSJW\\"><span class=\\"sc-bwzfXH sc-bxivhb hwMfcM\\">Call Center  <img src=\\"https://via.placeholder.com/50\\"/></span></div>"`;
